mmv1/third_party/terraform/framework_utils/framework_utils_test.go

```diff
@@ -50,3 +50,174 @@ func TestGetProjectFramework(t *testing.T) {
+					return
+				}
+				t.Fatalf("Got %d unexpected error(s) during test: %s", diags.ErrorsCount(), diags.Errors())
+			}
+
+			if region != tc.ExpectedRegion {
+				t.Fatalf("Incorrect region: got %s, want %s", region, tc.ExpectedRegion)
+			}
+		})
+	}
+}
+
+func TestGetZoneFramework(t *testing.T) {
+	cases := map[string]struct {
+		ResourceRegion types.String
+		ResourceZone   types.String
+		ProviderRegion types.String
+		ProviderZone   types.String
+		ExpectedZone   types.String
+		ExpectedError  bool
+	}{
+		"zone is pulled from the resource config": {
+			ResourceZone: types.StringValue("foo"),
+			ExpectedZone: types.StringValue("foo"),
+		},
+		"zone is pulled from the resource config's region value if available": {
+			ResourceZone: types.StringNull(),
+			ProviderZone: types.StringValue("bar"),
+			ExpectedZone: types.StringValue("bar"),
+		},
+		"error when zone is not set on the provider nor the resource": {
+			ExpectedError: true,
+		},
+	}
+	for tn, tc := range cases {
+		t.Run(tn, func(t *testing.T) {
+			// Arrange
+			var diags diag.Diagnostics
+
+			// Act
+			region := getZoneFramework(tc.ResourceZone, tc.ProviderZone, &diags)
+
+			// Assert
+			if diags.HasError() {
+				if tc.ExpectedError {
+					return
+				}
+				t.Fatalf("Got %d unexpected error(s) during test: %s", diags.ErrorsCount(), diags.Errors())
+			}
+
+			if region != tc.ExpectedZone {
+				t.Fatalf("Incorrect zone: got %s, want %s", region, tc.ExpectedZone)
+			}
+		})
+	}
+}
+
+func TestGetLocationFramework(t *testing.T) {
+	cases := map[string]struct {
+		ResourceLocation types.String
+		ResourceRegion   types.String
+		ResourceZone     types.String
+		ProviderZone     types.String
+		ExpectedLocation types.String
+		ExpectedError    bool
+	}{
+		"location is pulled from the resource config": {
+			ResourceLocation: types.StringValue("resource-location"),
+			ResourceRegion:   types.StringNull(),
+			ResourceZone:     types.StringNull(),
+			ProviderZone:     types.StringNull(),
+			ExpectedLocation: types.StringValue("resource-location"),
+		},
+		"region is pulled from the resource config when location is not set": {
+			ResourceLocation: types.StringNull(),
+			ResourceRegion:   types.StringValue("resource-region"),
+			ResourceZone:     types.StringNull(),
+			ProviderZone:     types.StringNull(),
+			ExpectedLocation: types.StringValue("resource-region"),
+		},
+		"zone is pulled from the resource config when location and region is not set": {
+			ResourceLocation: types.StringNull(),
+			ResourceRegion:   types.StringNull(),
+			ResourceZone:     types.StringValue("resource-zone"),
+			ProviderZone:     types.StringNull(),
+			ExpectedLocation: types.StringValue("resource-zone"),
+		},
+		//"zone pulled from the resource config can be retrieved by splitting on slashes and selecting last element": {
+		// This behaviour isn't implemented but is present in the SDK version of getLocation.
+		// Do we reproduce it? Or leave it behind?
+		//},
```

Comment:
I've asked in the team channel about this - do we want to replicate this behaviour?